### PR TITLE
Init CI build/test workflow for heapster

### PR DESCRIPTION
**Pull request by LisannaAtGalois**
Basic CI workflow for heapster on Mac + Linux which builds saw-script and then executes the examples in `deps/heapster-saw/examples`. ci-heapster.sh derived from ci.sh in master branch so future integration will be easy.

Example of what a test failure looks like: https://github.com/GaloisInc/saw-script/runs/1244155649

### CODE REVIEW

File: .github/workflows/heapster.yml
Comment (LisannaAtGalois) [inline .github/workflows/heapster.yml:32]: This line enables checkout of the branch without having any authentication setup. The heapster-saw submodule is specified using the SSH form of the URL, which requires you to be authenticated to GitHub via an SSH key to clone the repository: https://github.com/GaloisInc/saw-script/blob/wip-heapster/.gitmodules#L54

This line instructs git to treat all remotes as HTTP, allowing the CI to clone the submodules without having any special SSH stuff setup.

Merging wip-heapster into master with the submodule using the SSH format could cause problems for users with workflows that aren't expecting to have to authenticate with SSH just to clone the repository and pull submodules. I can yank this line so that the CI will fail in this case if this was an accidental inclusion and you want to continue guarding against it (CI in master will fail if wip-heapster is merged since it doesn't have a corresponding line like this). If you want the CI to not guard against this type of thing then I can also create a one-line patch against master to add this, so that merging wip-heapster into master won't cause a CI failure.

File: .github/ci-heapster.sh
Comment (kquick) [inline .github/ci-heapster.sh:10]: Shouldn't `$BIN` be prefixed to `$PATH` globally so that we can find/execute `$2` if it is in `$BIN`?  In that scenario, the first test here isn't needed, just the `command -v` check.

**Review by kquick — APPROVED** [review]
It's your call on whether you think the issues I noted should be fixed before merging.

File: .github/ci-heapster.sh
Comment (jared-w) [reply .github/ci-heapster.sh:119]: Because the github runners tend to be a bit flaky every now and then. The retry function is something I lifted verbatim from the rust language's CI setup a while ago and it's helped reduce spurious failures due to hiccups.

Comment (kquick) [reply .github/ci-heapster.sh:119]: OK.  From the CI perspective, that makes me cringe since builds should be reproducible and stable, but if it's needed for github runners I guess I'll just bite my tongue.

File: .github/ci-heapster.sh
Comment (jared-w) [reply .github/ci-heapster.sh:134]: This is true. We don't build this on windows currently because it doesn't compile correctly. The is_win check is a leftover from when I was still trying to build everything on windows and hadn't yet given up and replaced the windows case with return.

File: .github/workflows/heapster.yml
Comment (LisannaAtGalois) [reply .github/workflows/heapster.yml:14]: `8.8.4` and `8.6.5` are currently run on the master branch, so it would make sense to at least add `8.8.4` to make it match. @eddywestbrook would you like to have this additionally enabled?

**Comment by LisannaAtGalois** [comment]
@eddywestbrook this should be ready to go in as-is. Let me know if you'd like additional testing with examples from another repository enabled as part of this pull request, otherwise we can look at it as a followup.

Do note the execution time for this workflow varies between 30 minutes and 1 hour. The variance is due to frequent cache misses, which I believe are being caused by us hitting the upper limit of the repository's 5GB cache allowance. Let me know if this is OK for this branch, otherwise we can look at some possible solutions for mitigating this.

**Comment by LisannaAtGalois** [comment]
> It's your call on whether you think the issues I noted should be fixed before merging.

Since ci-heapster.sh is just a stand-in for ci.sh in the master branch and will get replaced, I don't think that making changes to it here would be beneficial.

File: .github/workflows/heapster.yml
Comment (LisannaAtGalois) [reply .github/workflows/heapster.yml:14]: Synced with Eddy on this and he's good with adding more versions, I'll enable that next.

**Pull request merged**
